# Model and Cost Function


## Model Representation

Como discutido nas aulas, a seguinte notação é utilizada a representação do problema:

* $m$: Quantidade de exemplos existentes;
* $x^{(i)}$: Representação de um exemplo/instância. O sobrescrito $i$ indica qual exemplo estamos nos referindo;
* $y^{(i)}$: Saída/resultado do modelo.

Dado um conjunto de dados disponível para a criação/treinamento de um modelo (no caso do estudo, um modelo de regressão linear de uma única variável), deseja-se gerar uma função h que aproxime os valores de entrada dos valores de saída. No caso da regressão linear, temos:

$$h_\theta(x) = \theta_0 + \theta_1x$$

onde $x$ é a variável de entrada, e $y = h_\theta (x) $.

Exemplificando um pouco os conceitos a partir do dataset do projeto da semana 2:

In [4]:
# Utilizando o pandas para a leitura do dataset.
import pandas as pd
dataset = pd.read_csv('../../data/week2/ex1data1.txt', header=None, names=['x', 'y'])

# observando as 5 primeiras amostras
dataset.head()


x        y
0  6.1101  17.5920
1  5.5277   9.1302
2  8.5186  13.6620
3  7.0032  11.8540
4  5.8598   6.8233

In [5]:
# Verificando a quantidade de dados existentes:
dataset.shape

(97, 2)

Considerando a tabela acima (ou dataframe), temos os seguintes exemplos, com a nomenclatura adequada:

* $m = 97$. Temos 97 exemplos no dataset;
* Para $i = 3$, temos $x^{(3)} = 8.5186$ e $y^{(3)} = 13.6620$. 

Como é possível notar, a notação está diferente. Segundo a proposta descrita pelo Andrew nas aulas, a contagem de $i$ se inicia em 1, ao contrário do que acontece no Python (em 0).

# Referências

* Model Representation: https://www.coursera.org/learn/machine-learning/supplement/cRa2m/model-representation